service/policy/attributes/attributes.go

```diff
@@ -410,18 +410,82 @@ func (s *AttributesService) RemoveKeyAccessServerFromValue(ctx context.Context,
 	return connect.NewResponse(rsp), nil
 }
 
-func (s *AttributesService) AssignKeyToAttribute(context.Context, *connect.Request[attributes.AssignKeyToAttributeRequest]) (*connect.Response[attributes.AssignKeyToAttributeResponse], error) {
-	return nil, connect.NewError(connect.CodeUnimplemented, nil)
+func (s *AttributesService) AssignKeyToAttribute(ctx context.Context, r *connect.Request[attributes.AssignKeyToAttributeRequest]) (*connect.Response[attributes.AssignKeyToAttributeResponse], error) {
+	rsp := &attributes.AssignKeyToAttributeResponse{}
+	auditParams := audit.PolicyEventParams{
+		ActionType: audit.ActionTypeCreate,
+		ObjectType: audit.ObjectTypeKasAttributeDefinitionKeyAssignment,
+	}
+
+	ak, err := s.dbClient.AssignPublicKeyToAttribute(ctx, r.Msg.GetAttributeKey())
+	if err != nil {
+		s.logger.Audit.PolicyCRUDFailure(ctx, auditParams)
+		return nil, db.StatusifyError(err, db.ErrTextCreationFailed, slog.String("attributeKey", r.Msg.GetAttributeKey().String()))
+	}
+
+	auditParams.ObjectID = ak.GetAttributeId()
+	s.logger.Audit.PolicyCRUDSuccess(ctx, auditParams)
+
+	rsp.AttributeKey = ak
+
+	return connect.NewResponse(rsp), nil
 }
 
-func (s *AttributesService) RemoveKeyFromAttribute(context.Context, *connect.Request[attributes.RemoveKeyFromAttributeRequest]) (*connect.Response[attributes.RemoveKeyFromAttributeResponse], error) {
-	return nil, connect.NewError(connect.CodeUnimplemented, nil)
+func (s *AttributesService) RemoveKeyFromAttribute(ctx context.Context, r *connect.Request[attributes.RemoveKeyFromAttributeRequest]) (*connect.Response[attributes.RemoveKeyFromAttributeResponse], error) {
+	rsp := &attributes.RemoveKeyFromAttributeResponse{}
+	auditParams := audit.PolicyEventParams{
+		ActionType: audit.ActionTypeDelete,
+		ObjectType: audit.ObjectTypeKasAttributeDefinitionKeyAssignment,
+	}
+
+	ak, err := s.dbClient.RemovePublicKeyFromAttribute(ctx, r.Msg.GetAttributeKey())
+	if err != nil {
+		s.logger.Audit.PolicyCRUDFailure(ctx, auditParams)
+		return nil, db.StatusifyError(err, db.ErrTextDeletionFailed, slog.String("attributeKey", r.Msg.GetAttributeKey().String()))
+	}
+
+	auditParams.ObjectID = ak.GetAttributeId()
+	s.logger.Audit.PolicyCRUDSuccess(ctx, auditParams)
+
+	return connect.NewResponse(rsp), nil
 }
 
-func (s *AttributesService) AssignKeyToValue(context.Context, *connect.Request[attributes.AssignKeyToValueRequest]) (*connect.Response[attributes.AssignKeyToValueResponse], error) {
-	return nil, connect.NewError(connect.CodeUnimplemented, nil)
+func (s *AttributesService) AssignKeyToValue(ctx context.Context, r *connect.Request[attributes.AssignKeyToValueRequest]) (*connect.Response[attributes.AssignKeyToValueResponse], error) {
```

Comment:
This is on me but in values I did AssignPublicKeyToValue. I feel like this should be updated to that.